### PR TITLE
Bug(ContextSelector) - Updated Param Type of toggleText in ContentSelector

### LINKED ISSUE #7457: Bug - ContextSelector - toggleText prop should allow for JSX
Right now the type is set to string, there is a use case to pass in text with an icon, so the type should be set to something like `ReactNode` instead.
https://github.com/patternfly/patternfly-react/blob/5bbf5776f7146b03c36382aa0e8de32781ded672/packages/react-core/src/components/ContextSelector/ContextSelector.tsx#L37

This issue came up in https://github.com/theforeman/foreman/pull/9222

### PR DESCRIPTION
**Pull request by adithyaakrishna**
Signed-off-by: Adithya Krishna <adikrish@redhat.com>

**What**: Closes #7457

**Additional issues**:

### CODE REVIEW

**Comment by nicolethoen** [comment]
@adithyaakrishna 
These proposed changes look good!
I have a few small notes and answers to your questions :)

- in the issue, you asked about changing the prop name, and that is something we are actually not supposed to do because that constitutes an API breaking change. So leaving it the way you have and simply updating the prop description in the comment above the prop is perfect!
- Could you also add an additional note in the prop description so that consumers are aware that the toggletext is placed in a button. Therefore, they should not include interactable elements in their toggleText to keep the html valid. Only non-interactive ReactNodes are valid.

**Comment by adithyaakrishna** [comment]
Hi @nicolethoen 
Apologies for the delay regarding this PR :(

I have added an additional note/desc for the prop as well, could you please let me know if it looks good?

File: packages/react-core/src/components/ContextSelector/ContextSelectorToggle.tsx [hunk 1]
```diff
@@ -10,8 +10,9 @@ export interface ContextSelectorToggleProps {
   id: string;
   /** Classes applied to root element of toggle */
   className?: string;
-  /** Text that appears in the Context Selector Toggle */
-  toggleText?: string;
+  /** Component or string that will be used in the Context Selector Toggle */
+  /** This prop is placed in a button */
```
Comment (nicolethoen) [inline packages/react-core/src/components/ContextSelector/ContextSelectorToggle.tsx:14]: This is great! 
Let's also be just a bit more explicit - maybe something like `This prop is placed in a button, and cannot contain any interactive elements`

And maybe both prop descriptions for `toggleText` should say the same thing. Could you update the description in `ContextSelector.tsx` as well?

**Review by nicolethoen — CHANGES_REQUESTED** [review]

File: packages/react-core/src/components/ContextSelector/ContextSelectorToggle.tsx (hunk 1 above)
Comment (adithyaakrishna) [reply packages/react-core/src/components/ContextSelector/ContextSelectorToggle.tsx]: Oops! My bad, I had missed updating it in the `ContextSelector.tsx`, thanks for pointing it out @nicolethoen :)

Comment (adithyaakrishna) [reply packages/react-core/src/components/ContextSelector/ContextSelectorToggle.tsx]: Updated the description with the suggestion as well 🚀

**Review by nicolethoen — APPROVED** [review]

**Review by nicolethoen — CHANGES_REQUESTED** [review]
Actually, looks like the linter isn't happy.

**Comment by adithyaakrishna** [comment]
@nicolethoen This is strange, husky missed this commit while I was updating the PR 🤔 

Fixed it in 774b5c7

File: packages/react-core/src/components/ContextSelector/ContextSelectorToggle.tsx [hunk 2]
```diff
@@ -32,7 +34,7 @@ export class ContextSelectorToggle extends React.Component<ContextSelectorToggle
   static displayName = 'ContextSelectorToggle';
   static defaultProps: PickOptional<ContextSelectorToggleProps> = {
     className: '',
-    toggleText: '',
+    toggleText: null as React.ReactNode,
```
Comment (adithyaakrishna) [inline packages/react-core/src/components/ContextSelector/ContextSelectorToggle.tsx:37]: Also @nicolethoen I don't think we will need this change, however would love to know your opinion on this 👀

Comment (nicolethoen) [reply packages/react-core/src/components/ContextSelector/ContextSelectorToggle.tsx:37]: I agree that it's likely unnecessary - an empty string is a valid value for a React.ReactNode

File: packages/react-core/src/components/ContextSelector/ContextSelector.tsx [hunk 3]
```diff
@@ -37,8 +37,10 @@ export interface ContextSelectorProps extends Omit<ToggleMenuBaseProps, 'menuApp
   isFullHeight?: boolean;
   /** Labels the Context Selector for Screen Readers */
   screenReaderLabel?: string;
-  /** Text that appears in the Context Selector Toggle */
-  toggleText?: string;
+  /** Component or string that will be used in the Context Selector.
+   * This prop is placed in a button, and cannot contain any interactive elements
+   */
+  toggleText?: React.ReactNode | string;
```
Comment (tlabaj) [inline packages/react-core/src/components/ContextSelector/ContextSelector.tsx:43]: You do not need the "string" here.  ReactNode includes string type.

File: packages/react-core/src/components/ContextSelector/ContextSelector.tsx [hunk 4]
```diff
@@ -75,7 +77,7 @@ export class ContextSelector extends React.Component<ContextSelectorProps, { oui
     onToggle: () => undefined as any,
     onSelect: () => undefined as any,
     screenReaderLabel: '',
-    toggleText: '',
+    toggleText: null as React.ReactNode,
```
Comment (tlabaj) [inline packages/react-core/src/components/ContextSelector/ContextSelector.tsx:80]: I  would keep the empty string default here.

File: packages/react-core/src/components/ContextSelector/ContextSelectorToggle.tsx [hunk 5]
```diff
@@ -10,8 +10,10 @@ export interface ContextSelectorToggleProps {
   id: string;
   /** Classes applied to root element of toggle */
   className?: string;
-  /** Text that appears in the Context Selector Toggle */
-  toggleText?: string;
+  /** Component or string that will be used in the Context Selector.
+   * This prop is placed in a button, and cannot contain any interactive elements
+   */
+  toggleText?: React.ReactNode | string;
```
Comment (tlabaj) [inline packages/react-core/src/components/ContextSelector/ContextSelectorToggle.tsx:16]: Sam comment as above.  ReactNode includes string type.

File: packages/react-core/src/components/ContextSelector/ContextSelectorToggle.tsx (hunk 2 above)
Comment (tlabaj) [inline packages/react-core/src/components/ContextSelector/ContextSelectorToggle.tsx:37]: You can keep the empty string default here.

**Review by tlabaj — CHANGES_REQUESTED** [review]

File: packages/react-core/src/components/ContextSelector/ContextSelectorToggle.tsx (hunk 2 above)
Comment (adithyaakrishna) [reply packages/react-core/src/components/ContextSelector/ContextSelectorToggle.tsx]: Oki Doki, updated the PR with the changes 🚀

File: packages/react-core/src/components/ContextSelector/ContextSelector.tsx (hunk 3 above)
Comment (adithyaakrishna) [reply packages/react-core/src/components/ContextSelector/ContextSelector.tsx]: Gotcha, I have updated the PR with the changes 👀

**Review by thatblindgeye — APPROVED** [review]

**Review by nicolethoen — APPROVED** [review]

File: packages/react-core/src/components/ContextSelector/ContextSelectorToggle.tsx [hunk 6]
```diff
@@ -10,8 +10,10 @@ export interface ContextSelectorToggleProps {
   id: string;
   /** Classes applied to root element of toggle */
   className?: string;
-  /** Text that appears in the Context Selector Toggle */
-  toggleText?: string;
+  /** Component or string that will be used in the Context Selector.
```
Comment (tlabaj) [inline packages/react-core/src/components/ContextSelector/ContextSelectorToggle.tsx:13]: Can this be sentence case please.  "Component or string that will be used in the context selector."

File: packages/react-core/src/components/ContextSelector/ContextSelector.tsx [hunk 7]
```diff
@@ -37,8 +37,10 @@ export interface ContextSelectorProps extends Omit<ToggleMenuBaseProps, 'menuApp
   isFullHeight?: boolean;
   /** Labels the Context Selector for Screen Readers */
   screenReaderLabel?: string;
-  /** Text that appears in the Context Selector Toggle */
-  toggleText?: string;
+  /** Component or string that will be used in the Context Selector.
```
Comment (tlabaj) [inline packages/react-core/src/components/ContextSelector/ContextSelector.tsx:40]: Can this be sentence case please. "Component or string that will be used in the context selector."

**Review by tlabaj — CHANGES_REQUESTED** [review]

File: packages/react-core/src/components/ContextSelector/ContextSelectorToggle.tsx (hunk 6 above)
Comment (adithyaakrishna) [reply packages/react-core/src/components/ContextSelector/ContextSelectorToggle.tsx]: Yes, I have updated it in the latest commit below :)

File: packages/react-core/src/components/ContextSelector/ContextSelector.tsx (hunk 7 above)
Comment (adithyaakrishna) [reply packages/react-core/src/components/ContextSelector/ContextSelector.tsx]: Done ✅

**Review by tlabaj — APPROVED** [review]

**Comment by adithyaakrishna** [comment]
Yayy, thanks @nicolethoen @tlabaj and @thatblindgeye 🚀

**Pull request merged**